Allow global password cache to be turned off

upsampled opened the pull request:
The global password cache is undesirable for higher security applications as a user's cache entry would remain even if the calling package deleted the corresponding `UsmSecurityParameters`.

Disabling this mechanism comes with a significant performance hit, as seen by running the added benchmark tests with the build tag `gosnmp_nopwdcache`.

There is also a middle ground approach, where the calling package can reset the cache by turning it off then on again. This should clear any stale entries and still allow the cache to be leveraged.

SuperQ commented on v3_usm_test.go:
```diff
@@ -228,3 +229,26 @@ func TestIsAuthenticaSHA512(t *testing.T) {
 }
+
+func BenchmarkSingleHash(b *testing.B) {
+	if len(os.Getenv("NOPWDCACHE")) > 0 {
```
We should use build tags not ENV vars for controlling tests.

upsampled replied:
As long as your ok with the extra file it is going to add in the top dir, that is fine with me

SuperQ replied:
Yes, new files are fine.

upsampled replied:
One alternative, adding `-pwdcache=false` (default true) as a testing argument. We already allow this in the `netsnmp` tests. Execution would look like:

```suggestion
go test -pwdcache=false -bench=.
```

upsampled replied:
Saves the need to create two extra files and is pretty clean. I also now realize i don't neet to call `flag.Parse` in the test so the changes are:

```go
var (
	pwdcache = flag.Bool("pwdcache", true, "enable or disable password caching")
)
```

and 
```go
func BenchmarkSingleHash(b *testing.B) {
	PasswordCaching(*pwdcache)
```

SuperQ replied:
I prefer build tags over test flags. What do you think @TimRots?

TimRots replied:
> I prefer build tags over test flags. What do you think @TimRots?

Lets be consistent and stick with build tags as you suggest.

upsampled replied:
Done in the below commit

TimRots reviewed (changes requested):
Almost there I think. Can we consider to make the build tag naming consistent with `gosnmp_nodebug`?  And add a short mention in `README.md:78` and `CHANGELOG.md` ?

upsampled commented:
@TimRots I changed the tag name. Regarding documentation, I added to the testing section of the README for `gosnmp_nopwdcache` because the tag only effects test behavior, unlike `gosnmp_nodebug` which changes behavior across the board.
